File: 0012-integer-to-roman/0012-integer-to-roman.go
import "strings"

type numeral struct {
	symbol string
	value  int
}

func intToRoman(num int) string {
	numerals := []numeral{
		{"M", 1000},
		{"CM", 900},
		{"D", 500},
		{"CD", 400},
		{"C", 100},
		{"XC", 90},
		{"L", 50},
		{"XL", 40},
		{"X", 10},
		{"IX", 9},
		{"V", 5},
		{"IV", 4},
		{"I", 1},
	}

	var res strings.Builder

	for _, numeral := range numerals {
		for num >= numeral.value {
			num -= numeral.value
			res.WriteString(numeral.symbol)
		}
	}

	return res.String()
}

File: 0012-integer-to-roman/0012-integer-to-roman.py
class Solution:

    value_symbols = (
        (1, "I"),
        (4, "IV"),
        (5, "V"),
        (9, "IX"),
        (10, "X"),
        (40, "XL"),
        (50, "L"),
        (90, "XC"),
        (100, "C"),
        (400, "CD"),
        (500, "D"),
        (900, "CM"),
        (1000, "M"),
    )

    def intToRoman(self, num: int) -> str:
        chars = []

        for val, char in Solution.value_symbols[::-1]:
            while num >= val:
                chars.append(char)
                num -= val

        return "".join(chars)